src/core/sig_state.ml

```diff
@@ -196,6 +192,21 @@ let open_sign : sig_state -> Sign.t -> sig_state = fun ss sign ->
   in
   {ss with in_scope; builtins; unops; binops; pp_hints}
 
+(** Dummy [sig_state] made from the dummy signature. *)
+let dummy : sig_state =
+  { signature = Sign.dummy (); in_scope = StrMap.empty; aliases = StrMap.empty
+  ; path_map = PathMap.empty; builtins = StrMap.empty; unops = StrMap.empty
+  ; binops = StrMap.empty; pp_hints = SymMap.empty }
+
+(** [of_sign sign] creates a state from the signature [sign] with ghost
+    signatures opened. *)
+let of_sign : Sign.t -> sig_state = fun sign ->
+  open_sign {dummy with signature = sign} Unif_rule.sign
+
+(** Dummy state made from the empty signature but with ghost modules
+    opened. *)
+let dummy : sig_state = of_sign (create_sign [])
```

Comment:
You do not remove it finally? (if not you should replace create_sign [] by Sign.dummy())

Reply:
We have to make a choice: `dummy` depends on `of_sign`, so I cannot use it in `of_sign`... or I can make a truly dummy signature which does not open the ghost modules (is that what you meant?)

Reply:
Yes, this is what I meant. There are Sign.dummy and Sig_state.dummy.